docs/DeveloperGuide.md

```diff
@@ -1,34 +1,978 @@
+  * Pros: Further increases efficiency for users.
+  * Cons: Increased complexity in implementation. 
+
+### 4.5 Feature: Rating system for books
+Given below is the class diagram for classes related to the Rating System in Quotesify:
+
+![Class Diagram for Rating system](images/ClassDiagram_Rating.png)
+
+* A `Book` must exists before a `Rating` can be created.
+* The `XRatingCommand` class represents `AddRatingCommand`, `DeleteRatingCommand`, `ListRatingCommand`
+, `EditRatingCommand`, `FindRatingCommand` classes which all extends their respective parent `XCommand` classes.
+* All the `XCommand` classes extends the abstract `Command` class.
+
+#### 4.5.1 Add rating
+The *add rating* feature will rely on an existing book object, and a rating object will then be created
+in the process.
+* The book object will store an attribute named *rating*, which will be set by this feature.
+* The rating object will be made up of the book object and a rating score, which is stored in a list of ratings.
+
+Given below is the sequence diagram for adding rating to a book:
+
+![Sequence Diagram for Add Ratings](images/SeqDiagram_AddRating.png)
+
+* The sequence diagram shows the process of *Add Rating* from the `execute()` method in `AddRatingCommand` class,
+which extends the `AddCommand` class. The switch statement in the `execute()` method to decide the item that the user
+is adding is not shown in the diagram.
+* The list of ratings will be retrieved from the `ListManager` class which stores all the different lists in Quotesify.
+* In the `addRating()` method, if the user input such as book title, author or rating score is missing
+, a message will be printed to inform the user and the method is returned.
+* There will also be checks implemented to check if the rating score is within range, if the book to be rated exists
+in Quotesify and if the book has been rated before. This is done by checking the list of books in Quotesify.
+If all these conditions are met, the book will be rated.
+* When rating a book, the attribute *rating* of the book is set to the rating score. A rating object containing the
+book details and rating score will also be created and stored in the rating list.
+This list of ratings will be used when listing or finding ratings.
+
+##### Design Consideration
+* Saving the ratings in a Rating List
+    * Pros: Helps in listing and finding ratings as not all books are rated.
+    * Cons: Increases memory usage.
+* Using both book title and author to identify a rating instead of just book title
+    * Pros: Allows books with the same title but different author to be rated.
+
+#### 4.5.2 Find ratings
+The *find ratings* feature will search if the rating for a particular book exists in Quotesify
+and print details about the rating.
+
+Given below is the sequence diagram for finding ratings:
+
+![Sequence Diagram for Find Ratings](images/SeqDiagram_FindRating.png)
```

Comment:
![Screenshot 2020-10-28 at 11 35 09 AM](https://user-images.githubusercontent.com/13443360/97387789-a4812e80-1911-11eb-8375-c63bda16ea7e.png)

Should the return line be simply like this? The "looping around" is not really correct UML format.